modules/auxiliary/scanner/ssh/ssh_erlangotp.rb

```diff
@@ -0,0 +1,183 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Erlang OTP Pre-Auth RCE Scanner',
+        'Description' => %q{
+          This module scans for CVE-2025-32433, a pre-authentication vulnerability in Erlang-based SSH servers
+          that allows remote command execution. It identifies vulnerable targets by connecting to the SSH service,
+          checking for an Erlang-specific banner, and sending a crafted packets to test the server's response.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Horizon3 Attack Team',
+          'Matt Keeley', # PoC
+          'Martin Kristiansen', # PoC
+          'mekhalleh (RAMELLA Sebastien)' # module author powered by EXA Reunion (https://www.exa.re/)
+        ],
+        'References' => [
+          ['CVE', '2025-32433'],
+          ['URL', 'https://x.com/Horizon3Attack/status/1912945580902334793'],
+          ['URL', 'https://platformsecurity.com/blog/CVE-2025-32433-poc'],
+          ['URL', 'https://github.com/ProDefense/CVE-2025-32433']
+        ],
+        'DisclosureDate' => '2025-04-16',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+  end
+
+  # Builds SSH_MSG_CHANNEL_OPEN for session
+  def build_channel_open(channel_id)
+    # SSH_MSG_CHANNEL_OPEN (0x5a) + formatted string + channel ID + window size + max packet size
```

Comment:
Metasploit uses the [hrr_rb_ssh](https://github.com/hirura/hrr_rb_ssh) libraries and also implements some [wrappers](https://github.com/rapid7/metasploit-framework/tree/master/lib/rex/proto/ssh) around it.

For example, the `SSH_MSG_CHANNEL_OPEN` structure is defined [here](https://github.com/hirura/hrr_rb_ssh/blob/master/lib/hrr_rb_ssh/messages/090_ssh_msg_channel_open.rb).

Do you think it would be possible to use our wrappers or hrr_rb_ssh instead of reimplementing everything? that would cleanup the code and make it easier to maintain.

Reply:
> Metasploit uses the [hrr_rb_ssh](https://github.com/hirura/hrr_rb_ssh) libraries and also implements some [wrappers](https://github.com/rapid7/metasploit-framework/tree/master/lib/rex/proto/ssh) around it.
> 
> For example, the `SSH_MSG_CHANNEL_OPEN` structure is defined [here](https://github.com/hirura/hrr_rb_ssh/blob/master/lib/hrr_rb_ssh/messages/090_ssh_msg_channel_open.rb).
> 
> Do you think it would be possible to use our wrappers or hrr_rb_ssh instead of reimplementing everything? that would cleanup the code and make it easier to maintain.

@cdelafuente-r7  do you have a concrete example of how using hrr_rb_ssh with the framework?
I'm getting errors when trying to access the library.